homeassistant/components/shelly/__init__.py

```diff
@@ -85,6 +85,7 @@ async def _async_update_data(self):
 
         try:
             async with async_timeout.timeout(5):
+                await self.device.update_status()
```

Comment:
We can't just put this here. 

In the current set up, Shelly devices will only be updated if we have an entity that needs it's data. With this approach we will always fetch both Rest + COAP data, regardless if an entity needs it. 

The Rest data of an entity will needs it's own update coordinator.